tools/cloudera-parcel-gcsconnector/create_parcel.sh

```diff
@@ -22,71 +22,183 @@
 # -f: name of the parcel in a single string format without any spaces or special characters.
 # -v: version of the parcel in the format x.x.x (ex: 1.0.0)
 # -o: name of the operating system distribution
-# -d: flag is to be used if you want to deploy the parcel to the cloudera manager parcel repo
+# -d: flag is to be used if you want to deploy the parcel to the Cloudera manager parcel repo
 # folder, this flag is optional and if not provided then the parcel file will be created in the
 # same directory where script run.
 
-while getopts f:v:o:d: option; do
+BASEDIR=${HOME}/GCS-CDH-PARCEL
+rm -rf ${BASEDIR}
+mkdir -p ${BASEDIR}
+LOGDIR=${HOME}/parcel-logs
+mkdir -p ${LOGDIR}/
+cd ${BASEDIR}
+touch ${LOGDIR}/cparcel_error.log
+touch ${LOGDIR}/cparcel_output.log
+
+#######################################
+# Capture logs in create_parcel.log and exit
+# Globals:
+#   None
+# Arguments:
+#   None
+# Returns:
+#   None
+#######################################
+function graceful_exit() {
+echo "-------------------------------">>${LOGDIR}/create_parcel.log
+echo "From User: "$(whoami)>>${LOGDIR}/create_parcel.log
+cat ${LOGDIR}/cparcel_output.log >> ${LOGDIR}/create_parcel.log
+exit 1
+}
+
+#######################################
+# User Help function for irregular input parameters
+# Globals:
+#   None
+# Arguments:
+#   -f: Parcel name
+#   -v: Parcel version
+#   -o: Operating system
+#   -d: (Optional)Parcel deployment through Clodera Manager value : true
+# Returns:
+#   None
+#######################################
+function help {
+  echo 'Usage: create_parcel.sh -f <parcel_name> -v <version> -o <os_distro_suffix> [-d true]'
+  graceful_exit
+}
+
+###Allowed OS parameter values and flag to check valid input
+os_flag=0
+os_value_arr=(el5 el6 el7 sles11 sles12 lucid precise trusty squeeze wheezy)
+
+###Link to hadoop jar file and flag to validate the file existence
+jar_link="https://storage.googleapis.com/hadoop-lib/gcs/gcs-connector-hadoop2-latest.jar"
+hadoop_jar_flag="1"
+
+exec > >(tee -i ${LOGDIR}/cparcel_output.log)
+exec 2>&1
+echo $(date)
+
+if [[ $# -lt 6 ]] || [[ $# == 7 ]] || [[ $# -gt 8 ]]; then
+  echo 'Error: Number of parameters do not match'
+  help
+  graceful_exit
+fi
+
+###*************
+#CHECKING The PARAMETERS
+###Regex varible for Parcel and version validity
+parcel_flag='^[A-Za-z0-9]+$'
+version_flag='^[0-9]\.[0-9]\.[0-9]$'
+
+### Check the validity of parameters at valid points
+if [[ $1 != '-f' || $3 != '-v' || $5 != '-o' ]]; then
```

Comment:
Perhaps this should be encapsulated in a validate parameters function.

Also should we be validating the parameters prior to parsing them? If we do the validation after the getopts we can do validation all in one pass and encapsulate everything in a single function.